src/care/LoopFuser.h

```diff
@@ -1254,13 +1254,23 @@ void LoopFuser<REGISTER_COUNT, XARGS...>::registerAction(const char * fileName,
 
 
 
-#define FUSIBLE_KERNEL_R_END return 0;}); }
+#define FUSIBLE_KERNEL_R_END return 0;}); } FUSIBLE_PHASE_RESET
 #define FUSIBLE_KERNEL_PHASE_R_END FUSIBLE_KERNEL_R_END
 #define FUSIBLE_KERNEL_END FUSIBLE_KERNEL_R_END
 
 #define FUSIBLE_KERNEL_PHASE(PRIORITY) FUSIBLE_KERNEL_PHASE_R(PRIORITY, CARE_DEFAULT_LOOP_FUSER_REGISTER_COUNT)
 
-#define FUSIBLE_PHASE_RESET FusedActionsObserver::getActiveObserver()->reset_phases(__FILE__, __LINE__);
+#define FUSIBLE_PHASE_RESET   if (FusedActions::flush_now) { \
+                                 for ( FusedActions *__fuser__ : { \
+                                    static_cast<FusedActions *> (LOOPFUSER(256)::getInstance()),\
+                                    static_cast<FusedActions *> (LOOPFUSER(128)::getInstance()),\
+                                    static_cast<FusedActions *> (LOOPFUSER(64)::getInstance()),\
+                                    FUSED_ACTION_INSTANCE_32 \
```

Comment:
Does `FusedActionsObserver::getActiveObserver()` also need to be accounted for here?

Reply:
FusedActionsObserver::getActiveObserver()->reset_phases(__FILE__, __LINE__);
calls flushActions if flush_now is set.

Reply:
I guess it's probably accounted for in the reset_phases call, but it would be nice for things to be consistent.